cargo-nextest/src/dispatch.rs

```diff
@@ -412,6 +434,30 @@ impl TestBuildFilter {
                 .map_err(|err| ExpectedError::CreateTestListError { err })?,
         )
     }
+
+    fn merge_test_binary_args(&mut self, test_binary_args: &mut Vec<String>) {
+        let mut ignore_filter = Vec::new();
+        test_binary_args.retain(|s| {
+            if s == "--include-ignored" {
+                ignore_filter.push(RunIgnored::All);
+                false
+            } else if s == "--ignored" {
+                ignore_filter.push(RunIgnored::IgnoredOnly);
+                false
+            } else if !s.starts_with('-') {
+                self.filter.push(s.to_owned());
+                false
+            } else {
+                true
+            }
+        });
+        for f in ignore_filter {
+            if self.run_ignored.is_some() && self.run_ignored != Some(f) {
+                log::warn!("skipping conflicting ignore filter: {}", f);
+            }
```

Comment:
Think it's best to error out entirely here so that `--run-ignored` and `-- --ignored` are treated as incompatible with each other.